paddle/fluid/framework/details/inplace_op_pass.cc

```diff
@@ -387,6 +381,20 @@ void InplacePass::ApplyImpl(ir::Graph *graph) const {
         continue;
       }
 
+      size_t in_arg_occur_times = all_in_args.count(in_arg);
+      if (in_arg_occur_times > 1) {
+        VLOG(4) << "Cannot inplace because Input(" << in_param << ")=" << in_arg
+                << " occurs " << in_arg_occur_times << " times in input of op "
+                << op_type;
+        continue;
+      }
+
+      size_t out_arg_occur_times = all_out_args.count(out_arg);
+      PADDLE_ENFORCE_EQ(
+          out_arg_occur_times, 1,
+          "Wrong graph: Output(%s)=%s occurs in other outputs of op %s",
+          out_param, out_arg, op_type);
```

Comment:
I am thinking that maybe it is not appropriate that adding the check here, it should be placed in program->graph.

Reply:
I agree.

Reply:
Done.